test/hummingbot/connector/exchange/bittrex/test_bittrex_order_book_data_source.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+import json
+import unittest
+from unittest.mock import AsyncMock, patch
+
+from hummingbot.connector.exchange.bittrex.bittrex_api_order_book_data_source import \
+    BittrexAPIOrderBookDataSource
+
+
+class BittrexOrderBookDataSourceTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.base_asset = "COINALPHA"
+        cls.quote_asset = "HBOT"
+        cls.trading_pair = f"{cls.base_asset}-{cls.quote_asset}"
+        cls.symbol = f"{cls.base_asset}{cls.quote_asset}"
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.ev_loop = asyncio.get_event_loop()
+
+        self.ws_incoming_messages = asyncio.Queue()
+        self.resume_test_event = asyncio.Event()
+        self._finalMessage = 'FinalDummyMessage'
+
+        self.output_queue = asyncio.Queue()
+
+        self.ob_data_source = BittrexAPIOrderBookDataSource(trading_pairs=[self.trading_pair])
+
+    def _create_queue_mock(self):
+        queue = AsyncMock()
+        queue.get.side_effect = self._get_next_ws_received_message
+        return queue
+
+    async def _get_next_ws_received_message(self):
+        message = await self.ws_incoming_messages.get()
+        if json.loads(message) == self._finalMessage:
+            self.resume_test_event.set()
+        return message
+
+    @patch("signalr_aio.Connection.start")
+    @patch("asyncio.Queue")
+    @patch(
+        "hummingbot.connector.exchange.bittrex.bittrex_api_order_book_data_source.BittrexAPIOrderBookDataSource"
+        "._transform_raw_message"
+    )
+    def test_listen_for_trades(self, transform_raw_message_mock, mocked_connection, _):
+        mocked_connection.return_value = self._create_queue_mock()
+        self.ws_incoming_messages.put_nowait(
+            json.dumps(
+                {
+                    'nonce': 1630292147820.41,
+                    'type': 'trade',
+                    'results': {
+                        'deltas': [
+                            {
+                                'id': 'b25fd775-bc1d-4f83-a82f-ff3022bb6982',
+                                'executedAt': '2021-08-30T02:55:47.75Z',
+                                'quantity': '0.01000000',
+                                'rate': '3197.61663059',
+                                'takerSide': 'SELL',
+                            }
+                        ],
+                        'sequence': 1228,
+                        'marketSymbol': self.trading_pair,
+                    }
+                }
+            )
+        )
+        transform_raw_message_mock.side_effect = lambda arg: json.loads(arg)
```

Comment:
Since we are patching the logic that parses messages, there is no point on transforming the dictionaries to JSON compliant strings. We could use the dictionaries all the time.

Reply:
Done